src/node_report_utils.cc

```diff
@@ -27,38 +54,17 @@ static void ReportEndpoints(uv_handle_t* h, JSONWriter* writer) {
     default:
       break;
   }
-  if (rc == 0) {
-    // uv_getnameinfo will format host and port and handle IPv4/IPv6.
-    uv_getnameinfo_t local;
-    rc = uv_getnameinfo(h->loop, &local, nullptr, addr, NI_NUMERICSERV);
-
-    if (rc == 0) {
-      writer->json_objectstart("localEndpoint");
-      writer->json_keyvalue("host", local.host);
-      writer->json_keyvalue("port", local.service);
-      writer->json_objectend();
-      wrote_local_endpoint = true;
-    }
+  if (rc != 0 || !ReportEndpoint(h, addr, "localEndpoint", writer)) {
+    writer->json_keyvalue("localEndpoint", null);
```

Comment:
Could you move this into `ReportEndpoint()`? Then, this logic could just be:

```c++
if (rc != 0)
  ReportEndpoint(h, addr, "localEndpoint", writer)
```

Same for the remote endpoint code below.